source/MRMesh/MRMeshOverhangs.cpp

```diff
@@ -0,0 +1,68 @@
+#include "MRMeshOverhangs.h"
+
+#include "MRBitSetParallelFor.h"
+#include "MRComputeBoundingBox.h"
+#include "MRExpandShrink.h"
+#include "MRMesh.h"
+#include "MRMeshComponents.h"
+#include "MRRegionBoundary.h"
+
+#include <cassert>
+
+namespace MR
+{
+
+std::vector<FaceBitSet> findOverhangs( const Mesh& mesh, const FindOverhangsSettings& settings )
+{
+    assert( std::abs( settings.axis.lengthSq() - 1.f ) < 1e-6f );
+    assert( settings.layerHeight > 0.f );
+    assert( settings.maxOverhangDistance > 0.f );
+    const auto minCos = -settings.layerHeight / std::hypot( settings.layerHeight, settings.maxOverhangDistance );
+
+    const auto isOverhanging = [&] ( FaceId f ) -> bool
+    {
+        const auto normal = mesh.normal( f );
+        const auto cos = dot( settings.axis, normal );
+        return cos < minCos;
+    };
+
+    // find faces that might create overhangs
+    FaceBitSet faces( mesh.topology.lastValidFace() + 1, false );
+    BitSetParallelFor( mesh.topology.getValidFaces(), [&] ( FaceId f )
+    {
+        faces[f] = isOverhanging( f );
+    } );
+
+    // simplify the regions...
+    expand( mesh.topology, faces, 1 );
```

Comment:
may be add a parameter for expand/shrink value?
or probably better even dilate/erode